core/src/kmx/kmx_xstring.cpp

```diff
@@ -107,6 +116,11 @@ km_core_cu *km::core::kmx::u16dup(km_core_cu *src) {
   memcpy(dup, src, (u16len(src) + 1) * sizeof(km_core_cu));
   return dup;
 }
+km_core_usv *km::core::kmx::u32dup(const km_core_usv *src) {
+  km_core_usv *dup = new km_core_usv[u32len(src) + 1];
+  memcpy(dup, src, (u32len(src) + 1) * sizeof(src[0]));
+  return dup;
+}
```

Comment:
So, the `.c_str()` returned by `std::u32string` is equivalent to `km_core_usv`?  Meaning it's std-string wrapper stuff that makes the difference between the two types?

Reply:
Though... I think that this means we're allocating new memory here that lacks an inherent memory management scheme.  I only see one new `free` call in this PR, and a new `delete` or two.  One of the `deletes` is in cleanup after a failed assertion, which is good, but I'm not having an easy time tracing all the codepaths that will need memory management based on this.

Reply:
This is patterned after C stdlib `strdup`. Caller is reponsible for memory management.

Reply:
Right and in both of the callers, the ICU and non-ICU, the buffer is deleted when it's done